src/hooks/use-webvtt.ts

```diff
@@ -89,11 +93,40 @@ const useWebVtt = () => {
     return cues.sort((cue1, cue2) => cue1.start - cue2.start);
   }
 
+  function parseVttData(data: string): Promise<Array<NodeWebVttCue>> {
+    return new Promise((resolve, reject) => {
+      const cues: Array<NodeWebVttCue> = [];
+      const parser = new WebVTT.Parser(window, WebVTT.StringDecoder());
+      parser.oncue = (cue: VTTCue) => {
+        const domTree: DocumentFragment = WebVTT.convertCueToDOMTree(
+          window,
+          cue.text,
+        );
+        const html = domTree.firstElementChild?.outerHTML || "&nbsp;";
+        const text = domTree.firstElementChild?.textContent || "";
+
+        cues.push({
+          identifier: uuidv4(),
+          start: cue.startTime,
+          end: cue.endTime,
+          align: cue.align,
+          html,
+          text,
+        });
+      };
```

Comment:
I'm seeing some initial issues where cues are not rendering as expected. Something in here seems to be not parsing correctly on the lunchroom manners recipe. 

![image](https://github.com/user-attachments/assets/bfbfd874-b381-46b7-a0d8-8a9e01d6281c)

Reply:
The VTT definitely loads and we get cues/time(s), but the text is coming back empty.